homeassistant/components/hafas/sensor.py

```diff
@@ -0,0 +1,169 @@
+"""Sensor for HaFAS."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+from enum import Enum
+from typing import Any
+
+from pyhafas import HafasClient
+from pyhafas.profile import DBProfile, VSNProfile
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA, SensorEntity
+from homeassistant.const import CONF_OFFSET
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+import homeassistant.util.dt as dt_util
+
+DOMAIN = "hafas"
+
+CONF_PROFILE = "profile"
+CONF_DESTINATION = "to"
+CONF_START = "from"
+DEFAULT_OFFSET = timedelta(minutes=0)
+CONF_ONLY_DIRECT = "only_direct"
+DEFAULT_ONLY_DIRECT = False
+
+ICON = "mdi:train"
+
+SCAN_INTERVAL = timedelta(minutes=2)
+
+
+class Profile(Enum):
+    """Enum of HaFAS profile type."""
+
+    DB = 0
+    VSN = 1
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

I suggest starting over from the beginning by running our scaffold script to get a solid base to build on.